cmd/juju/application/diffbundle.go

```diff
@@ -338,21 +338,13 @@ func (c *diffBundleCommand) bundleDataSource(ctx *cmd.Context, apiRoot base.APIC
 
 	bundleURL, bundleOrigin, err := charmAdaptor.ResolveBundleURL(bURL, origin)
 	if err != nil {
-		// Handle the charmhub failures during a resolving a bundle.
-		isCharmHub := charm.CharmHub.Matches(bURL.Schema)
-		if isCharmHub && errors.IsNotSupported(err) {
-			msg := `
-Current controller version is not compatible with CharmHub bundles.
-Consider using a CharmStore bundle instead.`
-			return nil, errors.Errorf(msg[1:])
-		} else if isCharmHub && errors.IsNotValid(err) {
+		if errors.Is(err, errors.NotValid) {
 			return nil, errors.Errorf("%q can not be found or is not a valid bundle", c.bundle)
```

Comment:
add `ctx.Verbosef(err)` - so the error isn't obvious, but we can find it as developers.